lib/secure_headers/headers/content_security_policy.rb

```diff
@@ -157,7 +157,7 @@ def dedup_source_list(sources)
       if wild_sources.any?
         sources.reject do |source|
           !wild_sources.include?(source) &&
-            wild_sources.any? { |pattern| File.fnmatch(pattern, source) }
+            wild_sources.any? { |pattern| URI(pattern).scheme == URI(source).scheme && File.fnmatch(pattern, source) }
```

Comment:
For large fields, this could result in the source URL being parsed a lot. Would it make sense to precompute the source scheme in the prior loop?